File: webapp/src/libs/semantic-kernel/model/AskResult.ts
// Copyright (c) Microsoft. All rights reserved.

//import { IChatMessage } from '../../models/ChatMessage';
import { ISearchValue } from '../../models/SearchResponse';

export interface IAskResult {
    value: string;
    variables: ContextVariable[];
}

export interface IAskSearchResult {
    count: number;
    value: ISearchValue[];
}

export interface ContextVariable {
    key: string;
    value: string;
}

File: webapi/Models/Response/AskResult.cs
// Copyright (c) Microsoft. All rights reserved.

namespace CopilotChat.WebApi.Models.Response;

public class AskResult
{
    public string Value { get; set; } = string.Empty;

    public IEnumerable<KeyValuePair<string, object?>>? Variables { get; set; } = Enumerable.Empty<KeyValuePair<string, object?>>();
}
